core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -51,6 +51,10 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
   private var _stack          = null.asInstanceOf[Array[Continuation]]
   private var _stackSize      = 0
   private val emptyTrace      = Trace.empty
+  private val nForkedFibers   = new AtomicInteger(0)
+
+  private[zio] def shouldYieldBeforeFork(): Boolean =
+    (nForkedFibers.incrementAndGet() % FiberRuntime.MaxForksBeforeYield) == 0
```

Comment:
Sorry if it's a dumb question 🙂

Something I don't understand with this mechanism is: how do you ensure this method is correctly called in all the necessary places? Just by discipline? 🤔 

I see that it's only called once in the private `ZIO#forkWithScopeOverride` method, which seems to be used by all the `.fork` variants.
What if someone makes a mistake in future evolutions and forgets to call this `shouldYieldBeforeFork` in a new/evolving version of a `fork` variant?

Could we find a way to better "connect" these 2 things/necessities so that no one can make an evolution that would forget to call `shouldYieldBeforeFork` and so forget to increment the `nForkedFibers` counter, which would lead to an incorrect counting of forked fibers? (Not that I'm not sure to be clear)
🤔

Reply:
I think this relates to this response https://github.com/zio/zio/pull/8745#discussion_r1568857696